src/endpoints/grpc/grpc.h

```diff
@@ -84,6 +33,13 @@ namespace ccf::grpc
           http::headervalues::contenttype::GRPC));
     }
 
+    // Set response header here rather than in set_grpc_response as data stream
+    // may be sent to client before endpoint returns
+    for (auto const& h : default_response_headers)
+    {
+      ctx->set_response_header(h.first, h.second);
+    }
+
```

Comment:
I don't think that's safe - what if the endpoint wants to overwrite these default headers? I think this is saying they might be able to, if they call `set_response_header` and overwrite this _before_ their endpoint calls something which triggers a data stream send? If that's the API, we need to make those send calls extremely obvious, and/or make header modification _after_ that send return an error.

Reply:
Good spot. This was only moved there before I added the `ccf::grpc::Stream` object so I've moved this back to `set_grpc_response`.